scripts/mkcloud

```diff
@@ -798,6 +798,11 @@ function onhost_reset_admin
     safely bootstrapcrowbar
 }
 
+function save_cloudsource
```

Comment:
Well the file `/etc/cloudsource` was only introduced to fix the start of tempest.

From the workflow perspective the only reliable variable is `$cloudsource`. During upgrades (depends on cloud version and on the fact if it was done manually stepping through the mkcloud steps) the variable `$upgrade_cloudsource` might not even be set in `qa_crowbarsetup.sh`. So we should not rely on it.

My proposal would be to clean up this mess like this:
- /etc/cloudsource is dropped
- we introduce a file like `/root/cloud.install.log` _(other short logs are in /root as well)_
- this file will be appended only with
  - `prepareinstcrowbar` appends the original installation cloud source: `installation: cloudsource=$cloudsource`
  - the `cloudupgrade` appends (only this function): `upgrade: cloudsource=$cloudsource`
  - we add a function to check if an upgrade is/was in progress that does this:
```
function is_cloud_upgrade
{
    grep -q "^upgrade: " ~/cloud.install.log
}
```

Other helpers might also be possible to figure out from where to where the upgrade was done.

With this we should be able to prevent the usage of `$upgrade_cloudsource` in qa_crowbarsetup and the resetting of `$cloudsource` to a value from a file entierly, because this is what makes the trouble.
Or put otherwise: `$cloudsource` is **only** changed right before an upgrade and at no other places. Workflow decisions should be done based on a grep on `/root/cloud.install.log` (and we will provide useful functions to prevent code duplication).

So with this I would even recommend to also close this PR - sorry.